homeassistant/components/hue.py

```diff
@@ -0,0 +1,237 @@
+        return
+
+    # If we came here and configuring this host, mark as done
+    if host in _CONFIGURING:
+        request_id = _CONFIGURING.pop(host)
+        configurator = get_component('configurator')
+        configurator.request_done(request_id)
+
+    # Add instanciated bridge to our own control global
+    if HUEBRIDGE.get(bridge_id):
+        _LOGGER.info('Bridge %s already initialized, skipping', bridge_id)
+        return
+    HUEBRIDGE[bridge_id] = HueBridge(hass, bridge)
+
+
+def request_configuration(hass, host):
+    """ Request configuration steps from the user. """
+    configurator = get_component('configurator')
+
+    # We got an error if this method is called while we are configuring
+    if host in _CONFIGURING:
+        configurator.notify_errors(
+            _CONFIGURING[host], "Failed to register, please try again.")
+        return
+
+    # pylint: disable=unused-argument
+    def hue_configuration_callback(data):
+        """ Actions to do when our configuration callback is called. """
+        setup_huebridge(hass, host)
+
+    _CONFIGURING[host] = configurator.request_config(
+        hass, "Philips Hue", hue_configuration_callback,
+        description=("Press the button on the bridge to register Philips Hue "
+                     "with Home Assistant."),
+        description_image="/static/images/config_philips_hue.jpg",
+        submit_caption="I have pressed the button"
+    )
+
+
+class HueBridge(object):
+    """
+    This manages a Hue bridge
+    Gets the latest data and update the states
+    """
+
+    def __init__(self, hass, bridge):
+        # This will be used to control devices from their platform modules
+        self.hass = hass
+        self._bridge = bridge
+
+        config = self._fetch_apidata().get('config')
+        self.bridge_id = config.get(ATTR_HUE_BRIDGEID)
+        self.bridge_ip = config.get(ATTR_HUE_BRIDGEIP)
+
+        # Support limitations in DiY zigbee bridge 'deconz'
+        # FIXME / This is based on the name, a user configureable parameter?
+        if config.get('name') == 'RaspBee-GW':
+            self.bridge_type = 'deconz'
+        else:
+            self.bridge_type = 'hue'
+
+        self.states = {}
+        self.update()
+
+        # FIXME / debuglog
+        _LOGGER.warning('New bridge: %s/%s', self.bridge_ip, self.bridge_id)
+
+    def set_light(self, light_id, command):
+        self._bridge.set_light(int(light_id), command)
+
+    def get_state(self, dev_type, dev_id):
+        try:
+            return self.states[dev_type][dev_id]
+        except ValueError:
+            return None
+
+    def _fetch_apidata(self):
+        """ Refresh state data from bridge """
+        try:
+            return  self._bridge.get_api()
+        except socket.error:
+            # socket.error when we cannot reach Hue
+            _LOGGER.error("Cannot reach the bridge")
+            return
+
+    #@Throttle(MIN_TIME_BETWEEN_UPDATES, MIN_TIME_BETWEEN_FORCED_UPDATES)
```

Comment:
I believe the Throttle timer is started when the object is instantiated. You can call this to force it to run:

```python
time.sleep(0.2)
self.update('no_throttle'=True)
```